### PR TITLE
Adds Tests-Serialization Package 

### PR DESCRIPTION
**Pull request by JoeAtHPI**
Includes 
- Example to Method Example
- Example to Method Test
- Example to Negative Method Test
- Tests \o/

Depends on https://github.com/hpi-swa-lab/babylonian-programming-smalltalk/pull/21

### CODE REVIEW

File: packages/Babylonian-Core.package/BPExampleToTest.class/instance/canBeCreated.in..st [hunk 1]
```diff
@@ -0,0 +1,9 @@
+helper
+canBeCreated: aMethodName in: aClass
```
Comment (codeZeilen) [inline packages/Babylonian-Core.package/BPExampleToTest.class/instance/canBeCreated.in..st:2]: This is interesting: This works as the combination of class, selector, and example name is unique. This is an assumption we might want to document here.

File: packages/Babylonian-Core.package/BPExampleToTest.class/instance/combineMethod.with..st [hunk 2]
```diff
@@ -0,0 +1,10 @@
+helper
+combineMethod: aMethodString with: anArrayOfParameters
+	"zips aMethodString together with anArrayOfParameters. 
+	example: combineMethod: 'do:with:' with: #(2 3) --> 'do: 2 with: 3'"
+	
+	| methodParts |
+	methodParts := (aMethodString splitBy: ':') allButLast.
+	^ methodParts isEmpty
```
Comment (codeZeilen) [inline packages/Babylonian-Core.package/BPExampleToTest.class/instance/combineMethod.with..st:8]: minor: Can also be done with #ifEmpty:ifNotEmpty:

File: packages/Babylonian-Core.package/BPExampleToTest.class/instance/exampleMethodClassName.st [hunk 3]
```diff
@@ -0,0 +1,4 @@
+private - serializing
+exampleMethodClassName
+
+	^ self example methodClass asString
```
Comment (codeZeilen) [inline packages/Babylonian-Core.package/BPExampleToTest.class/instance/exampleMethodClassName.st:4]: #asString -> #name

File: packages/Babylonian-Core.package/BPExampleToTest.class/instance/isGeneratedExampleTest.in..st [hunk 4]
```diff
@@ -0,0 +1,11 @@
+helper
```
Comment (codeZeilen) [inline packages/Babylonian-Core.package/BPExampleToTest.class/instance/isGeneratedExampleTest.in..st:1]: The empty lines could be removed

File: packages/Babylonian-Core.package/BPExampleToTest.class/instance/serializedTestName.st [hunk 5]
```diff
@@ -0,0 +1,4 @@
+private - serializing
+serializedTestName
```
Comment (codeZeilen) [inline packages/Babylonian-Core.package/BPExampleToTest.class/instance/serializedTestName.st:2]: Hm... I do not have a good overview of the relation of methods here but should that not also include the method selector?

File: packages/Babylonian-UI.package/BPExampleMorph.class/instance/setExecutionResultButtonColorAccordingToExpression..st [hunk 6]
```diff
@@ -0,0 +1,6 @@
+initialization
+setExecutionResultButtonColorAccordingToExpression: aBooleanExpression
```
Comment (codeZeilen) [inline packages/Babylonian-UI.package/BPExampleMorph.class/instance/setExecutionResultButtonColorAccordingToExpression..st:2]: The word expression is not right here. This is a plain Boolean object.

Comment (codeZeilen) [reply packages/Babylonian-UI.package/BPExampleMorph.class/instance/setExecutionResultButtonColorAccordingToExpression..st:2]: As a result, you might also want to shorten the message selector. :)

**Review by codeZeilen — COMMENTED** [review]
All of the comments are minor and you might as well ignore them if you want :)

General style comments:
* Superfluous parentheses
* Sometimes too many empty lines

File: packages/Babylonian-Core.package/BPExampleToTest.class/instance/serializedTestName.st (hunk 5 above)
Comment (Paulpanther) [reply packages/Babylonian-Core.package/BPExampleToTest.class/instance/serializedTestName.st:2]: The test name currently is: `test<name of example in camel case>` which is what this method does. What method selector do you mean?

File: packages/Babylonian-Core.package/BPExampleToMethodExample.class/class/canSerialize..st [hunk 7]
```diff
@@ -0,0 +1,5 @@
+handling
+canSerialize: anExample
+
+	"The user hasn't bothered to enter an expected result. So we assume they don't care about it."
+	^ anExample expectedResult isNil
```
[the remark this answers is not in the document]
Comment (Paulpanther) [reply packages/Babylonian-Core.package/BPExampleToMethodExample.class/class/canSerialize..st:5]: no. there are multiple serializes. this one can only serialize a method example and not a method test, therefore the exclusion of examples with expected result. The BPExampleToTestSerializer uses canSerialize to determine the right serializer for an example

File: packages/Babylonian-Core.package/BPExampleToMethodExample.class/README.md [hunk 8]
```diff
@@ -0,0 +1 @@
+Serializes an example by simply calling its method with given parameters. Not a test in a strict sense as assertions are missing, but a help to the user on how said method can be used.
```
[the remark this answers is not in the document]
Comment (Paulpanther) [reply packages/Babylonian-Core.package/BPExampleToMethodExample.class/README.md:1]: :+1:

**Pull request merged**